packages/ab-testing/src/index.ts

```diff
@@ -1,9 +1,76 @@
-// add another package with `yo lerna-typescript:package @myscope/my-new-package` instead of `lerna create`
+import crc32 from 'fast-crc32c';
 
-export function greet(name: string): string {
-    return `ab-testing says: hello to ${name}`;
+type HashObject = (
+    object: { [key: string]: string | number | undefined },
+    salt: string
+) => { [s: string]: string };
+
+type ForceInclude = {
+    [hashedKey: string]: string[];
+};
+
+type Cohort = {
+    name: string;
+    allocation?: [number, number][];
+    force_include?: ForceInclude;
+};
+
+type Experiment = {
+    name: string;
+    cohorts: Cohort[];
+};
+
+export type ABTestingConfig = {
+    salt: string;
+    version: '1.0';
+    experiments: Experiment[];
+};
+
+export type UserProfile = {
+    user_id: number;
+    user_type?: string;
+    email?: string;
+    email_domain?: string;
+};
+
+function getModuloValue(experiment: string, userId: number): number {
+    return crc32.calculate(String(userId), crc32.calculate(experiment)) % 100;
 }
 
-export function add(a: number, b: number): number {
-    return a + b;
+export class Experiments {
+    config: ABTestingConfig;
+    profile: UserProfile;
+    hashedProfile: ReturnType<HashObject>;

```

Comment:
This creates a dependency between the hashing and this interface; I don't think we should do this. Let's keep things separated and independent, and make them composable.

This interface should be:

```js
new Experiments(
    config,
    user.id,
    hashObject(user, config.salt)
)
```

That way we do not make hashing mandatory, and we do not create unnecessary dependencies.